weaver/db.R

```diff
@@ -19,46 +19,81 @@
 library(DBI)
 
 loadDBData <- function(connection) {
-
-    results <- dbGetQuery(connection, paste(
-    "SELECT used, quota, archived, last_modified, pi_id, unix_id, volume_id, record_date, warning_id
-    FROM ", conf$database, ".lustre_usage WHERE (record_date, volume_id) IN (
-        SELECT MAX(record_date), volume_id FROM ", conf$database, ".lustre_usage
-        GROUP BY volume_id)"), sep="")
-
-    results <- results  %>% 
-    left_join(pis, copy=TRUE)  %>% 
-    inner_join(unix_groups, by=c("unix_id" = "group_id"), copy=TRUE) %>%
+  dbGetQuery(
+    connection, 
+    glue::glue(
+      "
+      SELECT 
+        used, 
+        quota, 
+        last_modified, 
+        pi_id, 
+        unix_id, 
+        volume_id, 
+        base_directory_id,
+        directory_path,
+        record_date, 
+        warning_id 
+      FROM {conf$database}.lustre_usage 
+      LEFT JOIN base_directory 
+        USING (base_directory_id)   
+      WHERE (record_date, volume_id) 
+        IN (
+          SELECT MAX(record_date), volume_id FROM {conf$database}.lustre_usage
+          GROUP BY volume_id 
+          )      "
+    )
+  ) %>% 
+    left_join(pis, copy = TRUE)  %>% 
+    inner_join(unix_groups, by = c("unix_id" = "group_id"), copy = TRUE) %>%
     inner_join(volumes, copy = TRUE)  %>% 
     inner_join(warning_levels, copy = TRUE)  %>% 
     collect() %>%
     # converts columns imported as int64 to double, they play nicer with the rest of R
     mutate(
-        `quota` = as.double(`quota`),
-        `used` = as.double(`used`)
-        ) %>%
-    mutate("used_gib" = round(readBytes(used, "gb"), digits=2), "quota_gib" = round(readBytes(quota, "gb"), digits = 2))  %>% 
+      quota = as.double(quota),
+      used = as.double(used)
+    ) %>% 
+    mutate(
+      used_gib = round(readBytes(used, "gb"), digits = 2), 
+      quota_gib = round(readBytes(quota, "gb"), digits = 2)
+    ) %>% 
     # creates a quota column
     mutate(
-        quota_use = na_if(round(`used_gib` * 100/`quota_gib`, digits = 2), Inf),
-        `quota_gib` = na_if(`quota_gib`, 0)
-        ) %>%
-    mutate(is_humgen_yn = ifelse(is_humgen == 1, "Yes", "No"), archived_yn = ifelse(archived == 1, "Yes", "No"))
-
-    return(results)
-
+      quota_use = na_if(round(used_gib * 100/quota_gib, digits = 2), Inf),
+      quota_gib = na_if(quota_gib, 0)
+    ) %>% 
+    rowwise() %>% 
+    mutate(
+      path = fs::path_rel(directory_path, fs::path("/lustre", scratch_disk))
```

Comment:
Is there still a requirement for us to hardcode "/lustre" strings? Any way around this?